packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -130,6 +131,24 @@ export class FileStorage implements DStore {
       const { notes: _notes, errors: initErrors } = await this.initNotes();
       errors = errors.concat(initErrors);
       _notes.map((ent) => {
+        // Check for duplicate IDs when adding notes to the map
+        if (this.notes[ent.id] !== undefined) {
+          const duplicate = this.notes[ent.id];
+          errors.push(
+            new DendronError({
+              message: `Note ${ent.fname} in ${VaultUtils.getName(
+                ent.vault
+              )} has the same ID as ${duplicate.fname} in ${VaultUtils.getName(
+                duplicate.vault
+              )}`,
+              payload: [NoteUtils.toLogObj(ent), NoteUtils.toLogObj(duplicate)],
+              // Duplicate IDs break some features, but we shouldn't stop initialization completely
+              severity: ERROR_SEVERITY.MINOR,
+              code: StatusCodes.BAD_REQUEST,
```

Comment:
@kevinslin - on the topic of code, status.  I'm now confused - what HTTP status code is the server sending back to the client when we have multiple errors with different codes? 

500 `INTERNAL_SERVER_ERROR` seems more appropriate than `BAD_REQUEST` for this situation.  A client calling `init()` is not a bad request; the problem is that the note store is in a corrupted state.

Reply:
the reason its a client error is because the store/workspace is "client side"/user input. 

if a user was modifying a file and had bad frontmatter and we tried to update the engine, we would presumably respond with a 4xx response. 

that being said, its also fair to say this is a server error because loading existing notes "server side" failed when the user called "init".  

i'm good with classifying this as a 500